tron/mcp.py

```diff
@@ -53,30 +56,34 @@ def reconfigure(self, namespace=None):
             log.exception(f"reconfigure failure: {e.__class__.__name__}: {e}")
             raise e
 
-    def _load_config(self, reconfigure=False, namespace_to_reconfigure=None):
+    def _load_config(self, reconfigure=False, namespace_to_reconfigure=None, tron_start_time=None):
         """Read config data and apply it."""
         with self.state_watcher.disabled():
             self.apply_config(
                 self.config.load(),
                 reconfigure=reconfigure,
                 namespace_to_reconfigure=namespace_to_reconfigure,
+                tron_start_time=tron_start_time,
             )
 
-    def initial_setup(self):
+    def initial_setup(self, tron_start_time=None):
         """When the MCP is initialized the config is applied before the state.
         In this case jobs shouldn't be scheduled until the state is applied.
         """
-        self._load_config()
+        # The job schedules have been applied in first step
+        self._load_config(tron_start_time=tron_start_time)
+        # Now the jobs will actually get scheduled once the state for action runs are restored
         self.restore_state(
             actioncommand.create_action_runner_factory_from_config(
                 self.config.load().get_master().action_runner,
             ),
+            tron_start_time=tron_start_time,
         )
         # Any job with existing state would have been scheduled already. Jobs
         # without any state will be scheduled here.
         self.jobs.run_queue_schedule()
 
-    def apply_config(self, config_container, reconfigure=False, namespace_to_reconfigure=None):
+    def apply_config(self, config_container, reconfigure=False, namespace_to_reconfigure=None, tron_start_time=None):
```

Comment:
this will get called pretty frequently post-boot, so i don't think we want to pass through the boot time here